packages/dds/test-dds-utils/src/ddsSnapshotHarness.ts

```diff
@@ -0,0 +1,86 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "node:assert";
+import { existsSync, readFileSync, writeFileSync } from "node:fs";
+import path from "node:path";
+
+// Simple filter to avoid tests with a name that would accidentally be parsed as directory traversal or other confusing things.
+const nameCheck = new RegExp(/^[^"/\\]+$/);
+
+let currentTestName: string | undefined;
+let currentTestFile: string | undefined;
+
+/**
+ * A utility function for setting up a snapshot directory for tests.
+ * @param snapshotsFolder - The folder where snapshots are stored.
+ * @param dirPath - The directory within the snapshots folder where tests will store their snapshots with specific
+ * catogories. Defaults to root directory ("/").
+ * @internal
+ */
+export function useSnapshotDirectory(snapshotsFolder: string, dirPath: string = "/"): void {
```

Comment:
i'd prefer we curry this API to make the snapshotsFolder get specified once. This will make it easier to use in packages that take a lot of snapshots. E.g. usage might be:

```typescript
// this can be either as a top-level utility that gets exported, or within a single test suite.
const { useSubDirectory, takeSnapshot } = createSnapshotSuite(path.resolve(__dirname, "../../../src/test/snapshots"));
```

this will let you:

- restore additional validation checks that are in current `snapshotTools` file (such as `assert(normalizedDir.startsWith(snapshotsFolder));`)`
- verify that the snapshot folder being passed in is an absolute path